src/components/GlobalSearch/globalSearch.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Cognite AS
+import {
+  CogniteClient,
+  Asset,
+  CogniteEvent,
+  FileInfo,
+  Timeseries,
+  EventFilter,
+  AssetFilterProps,
+  FileFilterProps,
+  TimeseriesFilter,
+} from '@cognite/sdk';
+import flatten from 'lodash/flatten';
+
+export type SearchItem =
+  | { resource: Asset; resourceType: 'asset' }
+  | { resource: CogniteEvent; resourceType: 'event' }
+  | { resource: Timeseries; resourceType: 'timeSeries' }
+  | { resource: FileInfo; resourceType: 'file' };
+
+function withDiscriminator<
+  T extends Asset | CogniteEvent | Timeseries | FileInfo
+>(l: T[], resourceType: string): SearchItem[] {
+  return l.map(resource => ({ resource, resourceType } as SearchItem));
+}
+
+export type SearchForFilter = {
+  assets?: AssetFilterProps | null;
+  timeSeries?: TimeseriesFilter | null;
+  events?: EventFilter | null;
+  files?: FileFilterProps | null;
+};
+
+export function globalSearch(
+  client: CogniteClient,
+  query: string,
+  { assets, timeSeries, events, files }: SearchForFilter,
+  limitPerItem: number
+): Promise<SearchItem[]> {
+  const resolveEmptyIfNull = <T, U>(
+    f: (filter: U) => Promise<T[]>,
+    filter?: U | null
+  ) => {
+    if (filter != null) {
+      return f(filter);
+    } else {
+      return Promise.resolve([]);
+    }
+  };
+
+  const searchForAssets = (filter: AssetFilterProps) =>
```

Comment:
why we can't use here `async/await` syntax?

Reply:
Same argument as above